projects/common/src/external/external-url-navigator.ts

```diff
@@ -1,39 +1,46 @@
 import { Injectable } from '@angular/core';
 import { ActivatedRouteSnapshot, CanActivate } from '@angular/router';
 import { Observable, of } from 'rxjs';
-import { ExternalNavigationParams, NavigationService, NavigationType } from '../navigation/navigation.service';
+import {
+  ExternalNavigationPathParams,
+  ExternalNavigationWindowHandling,
+  NavigationService
+} from '../navigation/navigation.service';
 import { assertUnreachable } from '../utilities/lang/lang-utils';
 
 @Injectable({ providedIn: 'root' })
 export class ExternalUrlNavigator implements CanActivate {
   public constructor(private readonly navService: NavigationService) {}
 
   public canActivate(route: ActivatedRouteSnapshot): Observable<boolean> {
-    const encodedUrl = route.paramMap.get(ExternalNavigationParams.Url);
-    const navType = route.paramMap.has(ExternalNavigationParams.NavigationType)
-      ? (route.paramMap.get(ExternalNavigationParams.NavigationType) as NavigationType)
+    const encodedUrl = route.paramMap.get(ExternalNavigationPathParams.Url);
+    const windowHandling = route.paramMap.has(ExternalNavigationPathParams.WindowHandling)
```

Comment:
Some of these changes I have merged already. Might wanna update branch to get a better diff.

Reply:
Sorry had to do it as some tests were failing. Hope I didn't duplicate your work

Reply:
I just had my branch based off of yours. No overlapping work @anandtiwary :)